core_types/src/eav.rs

```diff
@@ -70,17 +97,33 @@ fn validate_attribute(attribute: &Attribute) -> HcResult<()> {
     }
 }
 
-impl EntityAttributeValue {
+impl EntityAttributeValueIndex {
     pub fn new(
         entity: &Entity,
         attribute: &Attribute,
         value: &Value,
-    ) -> HcResult<EntityAttributeValue> {
+    ) -> HcResult<EntityAttributeValueIndex> {
         validate_attribute(attribute)?;
-        Ok(EntityAttributeValue {
+        Ok(EntityAttributeValueIndex {
             entity: entity.clone(),
             attribute: attribute.clone(),
             value: value.clone(),
+            index: Utc::now().timestamp_nanos(),
+        })
+    }
+
+    pub fn new_with_timestamp(
```

Comment:
i don't like that we're mixing terminology like "timestamp" and "index"

should just be "index" everywhere

Reply:
I can change that for sure